spec/mailers/order_mailer_spec.rb

```diff
@@ -216,4 +216,41 @@
       end
     end
   end
+
+  describe "#invoice_email" do
+    subject(:email) { described_class.invoice_email(order) }
+    let(:order) { create(:completed_order_with_totals) }
+    let!(:invoice_data_generator){ InvoiceDataGenerator.new(order) }
+    let!(:invoice){
+      create(:invoice, order:,
+                       data: invoice_data_generator.serialize_for_invoice)
+    }
+
+    let(:generator){ double(:generator) }
+    let(:renderer){ double(:renderer) }
```

Comment:
I would use `instance_double` here, from the [doc](https://github.com/rspec/rspec-mocks/tree/main/features/verifying_doubles) : "Verifying doubles are a stricter alternative to [normal doubles](https://github.com/rspec/rspec-mocks/blob/main/features/verifying_doubles/basics/test-doubles) that provide guarantees about what is being verified."

Reply:
that's new for me
thank you for sharing :+1: